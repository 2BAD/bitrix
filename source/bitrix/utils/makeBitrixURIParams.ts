import { BitrixCommandParams } from '../types'

export default (params: BitrixCommandParams): string =>
  Object.keys(params)
    .reduce((query, name) => `${query}&${name}=${params[name]}`, '')
    // That will happen only for first instance
    .replace('&', '')
