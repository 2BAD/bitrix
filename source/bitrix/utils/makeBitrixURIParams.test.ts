/* eslint-env jest */
// tslint:disable: no-expression-statement

import makeBitrixURIParams from './makeBitrixURIParams'

describe('`makeBitrixURIParams` method', () => {
  it('should transform command params object into the query part of the Bitrix API URI', () => {
    expect(makeBitrixURIParams({ param1: 1, param2: 'value' })).toMatchSnapshot()
  })

  it('should return empty string when there is no params', () => {
    expect(makeBitrixURIParams({})).toMatchSnapshot()
  })
})
