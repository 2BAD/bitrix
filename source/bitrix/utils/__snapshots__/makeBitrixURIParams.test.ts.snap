// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`makeBitrixURIParams\` method should return empty string when there is no params 1`] = `""`;

exports[`\`makeBitrixURIParams\` method should transform command params object into the query part of the Bitrix API URI 1`] = `"param1=1&param2=value"`;
